[BUG] [cpp-pistache-server] Generated cmake project does not build

#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

<!--
Please follow the issue template below for bug reports.
Also please indicate in the issue title which language/library is concerned. Eg:  [BUG][JAVA] Bug generating foo with bar 
-->

##### Description

A recent [pull request](https://github.com/pistacheio/pistache/pull/870) in the pistache repository made changes to their build system, in which the `install` target was removed from the `CMakeLists.txt` (see [diff](https://github.com/pistacheio/pistache/commit/9dc080b9ebbe6fc1726b45e9db1550305938313e#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aL107) ).

That target is used by the pistache templates, and thus building the server will currently fail.
Tested on current master (7bf792a3485b967a2ae74f6b3bb3a337ba47e1bb).

Build commands used
```sh
cmake -S . -B build/
cmake --build build/ -j 1 --target all
```

Last bits of the output:
```
[ 16%] Performing build step for 'PISTACHE'
[ 91%] Built target pistache
[ 95%] Built target pistache_static
[100%] Built target pistache_shared
[ 20%] Performing install step for 'PISTACHE'
gmake[3]: *** No rule to make target 'install'.  Stop.
gmake[2]: *** [CMakeFiles/PISTACHE.dir/build.make:93: PISTACHE-prefix/src/PISTACHE-stamp/PISTACHE-install] Error 2
gmake[1]: *** [CMakeFiles/Makefile2:154: CMakeFiles/PISTACHE.dir/all] Error 2
gmake: *** [Makefile:103: all] Error 2
```


##### openapi-generator version

openapi-generator-cli 5.1.1-SNAPSHOT
commit : 7bf792a
built : 2021-04-01T11:56:22+02:00

##### OpenAPI declaration file content or url

The yaml was the sample from https://swagger.io/docs/specification/basic-structure/
```yaml
openapi: 3.0.0
info:
  title: Sample API
  description: Optional multiline or single-line description in [CommonMark](http://commonmark.org/help/) or HTML.
  version: 0.1.9

servers:
  - url: http://api.example.com/v1
    description: Optional server description, e.g. Main (production) server
  - url: http://staging-api.example.com
    description: Optional server description, e.g. Internal staging server for testing

paths:
  /users:
    get:
      summary: Returns a list of users.
      description: Optional extended description in CommonMark or HTML.
      responses:
        '200':    # status code
          description: A JSON array of user names
          content:
            application/json:
              schema: 
                type: array
                items: 
                  type: string
```

##### Generation Details

<!-- 
    Prefer CLI steps, including the language, libraries and various options. 
    Providing config-based settings allows for simpler testing across CLI and plugins. 
    For examples, see https://github.com/OpenAPITools/openapi-generator/tree/master/bin/configs
-->
```sh
java -jar modules/openapi-generator-cli/target/openapi-generator-cli.jar generate -g cpp-pistache-server -o out_temp2 -i ./sample.yaml
```

##### Steps to reproduce

<!-- unambiguous set of steps to reproduce the bug.-->
1. Clone and build latest master
2. Generate any cpp-pistache-server
3. Run cmake build

##### Related issues/PRs

<!-- has a similar issue/PR been reported/opened before? Please do a search in https://github.com/openapitools/openapi-generator/issues?utf8=%E2%9C%93&q=is%3Aissue%20 -->
N/A

##### Suggest a fix

<!-- if you can't fix the bug yourself, perhaps you can point to what might be
  causing the problem (line of code or commit), or simply make a suggestion -->
N/A

--- a/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CppPistacheServerCodegen.java
+++ b/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CppPistacheServerCodegen.java
@@ -290,14 +290,14 @@ public class CppPistacheServerCodegen extends AbstractCppCodegen {
 
                 //TODO: This changes the info about the real type but it is needed to parse the header params
                 if (param.isHeaderParam) {
-                    param.dataType = "Pistache::Optional<Pistache::Http::Header::Raw>";
-                    param.baseType = "Pistache::Optional<Pistache::Http::Header::Raw>";
+                    param.dataType = "std::optional<Pistache::Http::Header::Raw>";
+                    param.baseType = "std::optional<Pistache::Http::Header::Raw>";
                 } else if (param.isQueryParam) {
                     if (param.isPrimitiveType) {
-                        param.dataType = "Pistache::Optional<" + param.dataType + ">";
+                        param.dataType = "std::optional<" + param.dataType + ">";
                     } else {
-                        param.dataType = "Pistache::Optional<" + param.dataType + ">";
-                        param.baseType = "Pistache::Optional<" + param.baseType + ">";
+                        param.dataType = "std::optional<" + param.dataType + ">";
+                        param.baseType = "std::optional<" + param.baseType + ">";
                     }
                 }
             }

--- a/modules/openapi-generator/src/main/resources/cpp-pistache-server/api-header.mustache
+++ b/modules/openapi-generator/src/main/resources/cpp-pistache-server/api-header.mustache
@@ -12,8 +12,8 @@
 #include <pistache/http.h>
 #include <pistache/router.h>
 #include <pistache/http_headers.h>
-#include <pistache/optional.h>
 {{^hasModelImport}}#include <nlohmann/json.hpp>{{/hasModelImport}}
+#include <optional>
 #include <utility>
 
 {{#imports}}{{{import}}}

--- a/modules/openapi-generator/src/main/resources/cpp-pistache-server/api-impl-header.mustache
+++ b/modules/openapi-generator/src/main/resources/cpp-pistache-server/api-impl-header.mustache
@@ -14,10 +14,10 @@
 #include <pistache/http.h>
 #include <pistache/router.h>
 #include <memory>
+#include <optional>
 
 #include <{{classname}}.h>
 
-#include <pistache/optional.h>
 
 {{#imports}}{{{import}}}
 {{/imports}}

--- a/modules/openapi-generator/src/main/resources/cpp-pistache-server/api-source.mustache
+++ b/modules/openapi-generator/src/main/resources/cpp-pistache-server/api-source.mustache
@@ -74,11 +74,11 @@ void {{classname}}::{{operationIdSnakeCase}}_handler(const Pistache::Rest::Reque
     // Getting the query params
     {{#queryParams}}
     auto {{paramName}}Query = request.query().get("{{baseName}}");
-    Pistache::Optional<{{^isContainer}}{{dataType}}{{/isContainer}}{{#isArray}}std::vector<{{items.baseType}}>{{/isArray}}> {{paramName}};
-    if(!{{paramName}}Query.isEmpty()){
+    std::optional<{{^isContainer}}{{dataType}}{{/isContainer}}{{#isArray}}std::vector<{{items.baseType}}>{{/isArray}}> {{paramName}};
+    if({{paramName}}Query.has_value()){
         {{^isContainer}}{{dataType}}{{/isContainer}}{{#isArray}}std::vector<{{items.baseType}}>{{/isArray}} valueQuery_instance;
-        if(fromStringValue({{paramName}}Query.get(), valueQuery_instance)){
-            {{paramName}} = Pistache::Some(valueQuery_instance);
+        if(fromStringValue({{paramName}}Query.value(), valueQuery_instance)){
+            {{paramName}} = valueQuery_instance;
         }
     }
     {{/queryParams}}

--- a/modules/openapi-generator/src/main/resources/cpp-pistache-server/cmake.mustache
+++ b/modules/openapi-generator/src/main/resources/cpp-pistache-server/cmake.mustache
@@ -17,7 +17,7 @@ ExternalProject_Add(PISTACHE
 
 ExternalProject_Add(NLOHMANN
     GIT_REPOSITORY https://github.com/nlohmann/json.git
-    CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${EXTERNAL_INSTALL_LOCATION}
+    CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${EXTERNAL_INSTALL_LOCATION} -DJSON_BuildTests=OFF
 )
 
 include_directories(${EXTERNAL_INSTALL_LOCATION}/include)

--- a/modules/openapi-generator/src/main/resources/cpp-pistache-server/cmake.mustache
+++ b/modules/openapi-generator/src/main/resources/cpp-pistache-server/cmake.mustache
@@ -2,7 +2,7 @@ cmake_minimum_required (VERSION 3.2)
 
 project(api-server)
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14 -pg -g3" )
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17 -pg -g3" )
 
 {{#addExternalLibs}}
 include(ExternalProject)

--- a/modules/openapi-generator/src/main/resources/cpp-pistache-server/cmake.mustache
+++ b/modules/openapi-generator/src/main/resources/cpp-pistache-server/cmake.mustache
@@ -10,15 +10,16 @@ include(ExternalProject)
 set(EXTERNAL_INSTALL_LOCATION ${CMAKE_CURRENT_SOURCE_DIR}/external)
 
 ExternalProject_Add(PISTACHE
-    GIT_REPOSITORY https://github.com/oktal/pistache.git
-    CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${EXTERNAL_INSTALL_LOCATION}
+    GIT_REPOSITORY https://github.com/pistacheio/pistache.git
+    BUILD_IN_SOURCE true
+    INSTALL_COMMAND meson setup build --prefix=${EXTERNAL_INSTALL_LOCATION} --libdir=lib && meson install -C build
 )
 
 ExternalProject_Add(NLOHMANN
     GIT_REPOSITORY https://github.com/nlohmann/json.git
     CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${EXTERNAL_INSTALL_LOCATION}
-
 )
+
 include_directories(${EXTERNAL_INSTALL_LOCATION}/include)
 link_directories(${EXTERNAL_INSTALL_LOCATION}/lib)
 {{/addExternalLibs}}

--- a/modules/openapi-generator/src/main/resources/cpp-pistache-server/model-struct-header.mustache
+++ b/modules/openapi-generator/src/main/resources/cpp-pistache-server/model-struct-header.mustache
@@ -12,7 +12,7 @@
 {{#imports}}{{{this}}}
 {{/imports}}
 #include <nlohmann/json.hpp>
-{{#hasOptional}}#include <pistache/optional.h>{{/hasOptional}}
+{{#hasOptional}}#include <optional>{{/hasOptional}}
 
 namespace {{modelNamespace}}
 {
@@ -20,7 +20,7 @@ namespace {{modelNamespace}}
 struct {{classname}}
 {
     {{#vars}}
-    {{^required}}Pistache::Optional<{{/required}}{{{dataType}}}{{^required}}>{{/required}} {{name}};
+    {{^required}}std::optional<{{/required}}{{{dataType}}}{{^required}}>{{/required}} {{name}};
     {{/vars}}
 
     bool operator==(const {{classname}}& other) const;

--- a/modules/openapi-generator/src/main/resources/cpp-pistache-server/model-struct-source.mustache
+++ b/modules/openapi-generator/src/main/resources/cpp-pistache-server/model-struct-source.mustache
@@ -45,7 +45,7 @@ void from_json(const nlohmann::json& j, {{classname}}& o)
     {{^required}}if (j.find("{{baseName}}") != j.end()) {
         {{{dataType}}} temporary_{{name}};
         j.at("{{baseName}}").get_to(temporary_{{name}});
-        o.{{name}} = Pistache::Some(temporary_{{name}});
+        o.{{name}} = temporary_{{name}};
     }{{/required}}
     {{/vars}}
 }

--- a/samples/server/petstore/cpp-pistache/CMakeLists.txt
+++ b/samples/server/petstore/cpp-pistache/CMakeLists.txt
@@ -2,22 +2,23 @@ cmake_minimum_required (VERSION 3.2)
 
 project(api-server)
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14 -pg -g3" )
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17 -pg -g3" )
 
 include(ExternalProject)
 
 set(EXTERNAL_INSTALL_LOCATION ${CMAKE_CURRENT_SOURCE_DIR}/external)
 
 ExternalProject_Add(PISTACHE
-    GIT_REPOSITORY https://github.com/oktal/pistache.git
-    CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${EXTERNAL_INSTALL_LOCATION}
+    GIT_REPOSITORY https://github.com/pistacheio/pistache.git
+    BUILD_IN_SOURCE true
+    INSTALL_COMMAND meson setup build --prefix=${EXTERNAL_INSTALL_LOCATION} --libdir=lib && meson install -C build
 )
 
 ExternalProject_Add(NLOHMANN
     GIT_REPOSITORY https://github.com/nlohmann/json.git
-    CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${EXTERNAL_INSTALL_LOCATION}
-
+    CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${EXTERNAL_INSTALL_LOCATION} -DJSON_BuildTests=OFF
 )
+
 include_directories(${EXTERNAL_INSTALL_LOCATION}/include)
 link_directories(${EXTERNAL_INSTALL_LOCATION}/lib)
 

--- a/samples/server/petstore/cpp-pistache/api/PetApi.cpp
+++ b/samples/server/petstore/cpp-pistache/api/PetApi.cpp
@@ -128,11 +128,11 @@ void PetApi::find_pets_by_status_handler(const Pistache::Rest::Request &request,
 
     // Getting the query params
     auto statusQuery = request.query().get("status");
-    Pistache::Optional<std::vector<std::string>> status;
-    if(!statusQuery.isEmpty()){
+    std::optional<std::vector<std::string>> status;
+    if(statusQuery.has_value()){
         std::vector<std::string> valueQuery_instance;
-        if(fromStringValue(statusQuery.get(), valueQuery_instance)){
-            status = Pistache::Some(valueQuery_instance);
+        if(fromStringValue(statusQuery.value(), valueQuery_instance)){
+            status = valueQuery_instance;
         }
     }
     
@@ -158,11 +158,11 @@ void PetApi::find_pets_by_tags_handler(const Pistache::Rest::Request &request, P
 
     // Getting the query params
     auto tagsQuery = request.query().get("tags");
-    Pistache::Optional<std::vector<std::string>> tags;
-    if(!tagsQuery.isEmpty()){
+    std::optional<std::vector<std::string>> tags;
+    if(tagsQuery.has_value()){
         std::vector<std::string> valueQuery_instance;
-        if(fromStringValue(tagsQuery.get(), valueQuery_instance)){
-            tags = Pistache::Some(valueQuery_instance);
+        if(fromStringValue(tagsQuery.value(), valueQuery_instance)){
+            tags = valueQuery_instance;
         }
     }
     

--- a/samples/server/petstore/cpp-pistache/api/PetApi.h
+++ b/samples/server/petstore/cpp-pistache/api/PetApi.h
@@ -22,8 +22,8 @@
 #include <pistache/http.h>
 #include <pistache/router.h>
 #include <pistache/http_headers.h>
-#include <pistache/optional.h>
 
+#include <optional>
 #include <utility>
 
 #include "ApiResponse.h"
@@ -86,7 +86,7 @@ private:
     /// </remarks>
     /// <param name="petId">Pet id to delete</param>
     /// <param name="apiKey"> (optional, default to &quot;&quot;)</param>
-    virtual void delete_pet(const int64_t &petId, const Pistache::Optional<Pistache::Http::Header::Raw> &apiKey, Pistache::Http::ResponseWriter &response) = 0;
+    virtual void delete_pet(const int64_t &petId, const std::optional<Pistache::Http::Header::Raw> &apiKey, Pistache::Http::ResponseWriter &response) = 0;
     /// <summary>
     /// Finds Pets by status
     /// </summary>
@@ -94,7 +94,7 @@ private:
     /// Multiple status values can be provided with comma separated strings
     /// </remarks>
     /// <param name="status">Status values that need to be considered for filter</param>
-    virtual void find_pets_by_status(const Pistache::Optional<std::vector<std::string>> &status, Pistache::Http::ResponseWriter &response) = 0;
+    virtual void find_pets_by_status(const std::optional<std::vector<std::string>> &status, Pistache::Http::ResponseWriter &response) = 0;
     /// <summary>
     /// Finds Pets by tags
     /// </summary>
@@ -102,7 +102,7 @@ private:
     /// Multiple tags can be provided with comma separated strings. Use tag1, tag2, tag3 for testing.
     /// </remarks>
     /// <param name="tags">Tags to filter by</param>
-    virtual void find_pets_by_tags(const Pistache::Optional<std::vector<std::string>> &tags, Pistache::Http::ResponseWriter &response) = 0;
+    virtual void find_pets_by_tags(const std::optional<std::vector<std::string>> &tags, Pistache::Http::ResponseWriter &response) = 0;
     /// <summary>
     /// Find pet by ID
     /// </summary>

--- a/samples/server/petstore/cpp-pistache/api/StoreApi.h
+++ b/samples/server/petstore/cpp-pistache/api/StoreApi.h
@@ -22,8 +22,8 @@
 #include <pistache/http.h>
 #include <pistache/router.h>
 #include <pistache/http_headers.h>
-#include <pistache/optional.h>
 
+#include <optional>
 #include <utility>
 
 #include "Order.h"

--- a/samples/server/petstore/cpp-pistache/api/UserApi.cpp
+++ b/samples/server/petstore/cpp-pistache/api/UserApi.cpp
@@ -211,19 +211,19 @@ void UserApi::login_user_handler(const Pistache::Rest::Request &request, Pistach
 
     // Getting the query params
     auto usernameQuery = request.query().get("username");
-    Pistache::Optional<std::string> username;
-    if(!usernameQuery.isEmpty()){
+    std::optional<std::string> username;
+    if(usernameQuery.has_value()){
         std::string valueQuery_instance;
-        if(fromStringValue(usernameQuery.get(), valueQuery_instance)){
-            username = Pistache::Some(valueQuery_instance);
+        if(fromStringValue(usernameQuery.value(), valueQuery_instance)){
+            username = valueQuery_instance;
         }
     }
     auto passwordQuery = request.query().get("password");
-    Pistache::Optional<std::string> password;
-    if(!passwordQuery.isEmpty()){
+    std::optional<std::string> password;
+    if(passwordQuery.has_value()){
         std::string valueQuery_instance;
-        if(fromStringValue(passwordQuery.get(), valueQuery_instance)){
-            password = Pistache::Some(valueQuery_instance);
+        if(fromStringValue(passwordQuery.value(), valueQuery_instance)){
+            password = valueQuery_instance;
         }
     }
     

--- a/samples/server/petstore/cpp-pistache/api/UserApi.h
+++ b/samples/server/petstore/cpp-pistache/api/UserApi.h
@@ -22,8 +22,8 @@
 #include <pistache/http.h>
 #include <pistache/router.h>
 #include <pistache/http_headers.h>
-#include <pistache/optional.h>
 
+#include <optional>
 #include <utility>
 
 #include "User.h"
@@ -118,7 +118,7 @@ private:
     /// </remarks>
     /// <param name="username">The user name for login</param>
     /// <param name="password">The password for login in clear text</param>
-    virtual void login_user(const Pistache::Optional<std::string> &username, const Pistache::Optional<std::string> &password, Pistache::Http::ResponseWriter &response) = 0;
+    virtual void login_user(const std::optional<std::string> &username, const std::optional<std::string> &password, Pistache::Http::ResponseWriter &response) = 0;
     /// <summary>
     /// Logs out current logged in user session
     /// </summary>

--- a/samples/server/petstore/cpp-pistache/impl/PetApiImpl.cpp
+++ b/samples/server/petstore/cpp-pistache/impl/PetApiImpl.cpp
@@ -27,13 +27,13 @@ PetApiImpl::PetApiImpl(const std::shared_ptr<Pistache::Rest::Router>& rtr)
 void PetApiImpl::add_pet(const Pet &body, Pistache::Http::ResponseWriter &response) {
     response.send(Pistache::Http::Code::Ok, "Do some magic\n");
 }
-void PetApiImpl::delete_pet(const int64_t &petId, const Pistache::Optional<Pistache::Http::Header::Raw> &apiKey, Pistache::Http::ResponseWriter &response) {
+void PetApiImpl::delete_pet(const int64_t &petId, const std::optional<Pistache::Http::Header::Raw> &apiKey, Pistache::Http::ResponseWriter &response) {
     response.send(Pistache::Http::Code::Ok, "Do some magic\n");
 }
-void PetApiImpl::find_pets_by_status(const Pistache::Optional<std::vector<std::string>> &status, Pistache::Http::ResponseWriter &response) {
+void PetApiImpl::find_pets_by_status(const std::optional<std::vector<std::string>> &status, Pistache::Http::ResponseWriter &response) {
     response.send(Pistache::Http::Code::Ok, "Do some magic\n");
 }
-void PetApiImpl::find_pets_by_tags(const Pistache::Optional<std::vector<std::string>> &tags, Pistache::Http::ResponseWriter &response) {
+void PetApiImpl::find_pets_by_tags(const std::optional<std::vector<std::string>> &tags, Pistache::Http::ResponseWriter &response) {
     response.send(Pistache::Http::Code::Ok, "Do some magic\n");
 }
 void PetApiImpl::get_pet_by_id(const int64_t &petId, Pistache::Http::ResponseWriter &response) {

--- a/samples/server/petstore/cpp-pistache/impl/PetApiImpl.h
+++ b/samples/server/petstore/cpp-pistache/impl/PetApiImpl.h
@@ -24,10 +24,10 @@
 #include <pistache/http.h>
 #include <pistache/router.h>
 #include <memory>
+#include <optional>
 
 #include <PetApi.h>
 
-#include <pistache/optional.h>
 
 #include "ApiResponse.h"
 #include "Pet.h"
@@ -44,9 +44,9 @@ public:
     ~PetApiImpl() override = default;
 
     void add_pet(const Pet &body, Pistache::Http::ResponseWriter &response);
-    void delete_pet(const int64_t &petId, const Pistache::Optional<Pistache::Http::Header::Raw> &apiKey, Pistache::Http::ResponseWriter &response);
-    void find_pets_by_status(const Pistache::Optional<std::vector<std::string>> &status, Pistache::Http::ResponseWriter &response);
-    void find_pets_by_tags(const Pistache::Optional<std::vector<std::string>> &tags, Pistache::Http::ResponseWriter &response);
+    void delete_pet(const int64_t &petId, const std::optional<Pistache::Http::Header::Raw> &apiKey, Pistache::Http::ResponseWriter &response);
+    void find_pets_by_status(const std::optional<std::vector<std::string>> &status, Pistache::Http::ResponseWriter &response);
+    void find_pets_by_tags(const std::optional<std::vector<std::string>> &tags, Pistache::Http::ResponseWriter &response);
     void get_pet_by_id(const int64_t &petId, Pistache::Http::ResponseWriter &response);
     void update_pet(const Pet &body, Pistache::Http::ResponseWriter &response);
     void update_pet_with_form(const Pistache::Rest::Request &request, Pistache::Http::ResponseWriter &response);

--- a/samples/server/petstore/cpp-pistache/impl/StoreApiImpl.h
+++ b/samples/server/petstore/cpp-pistache/impl/StoreApiImpl.h
@@ -24,10 +24,10 @@
 #include <pistache/http.h>
 #include <pistache/router.h>
 #include <memory>
+#include <optional>
 
 #include <StoreApi.h>
 
-#include <pistache/optional.h>
 
 #include "Order.h"
 #include <map>

--- a/samples/server/petstore/cpp-pistache/impl/UserApiImpl.cpp
+++ b/samples/server/petstore/cpp-pistache/impl/UserApiImpl.cpp
@@ -39,7 +39,7 @@ void UserApiImpl::delete_user(const std::string &username, Pistache::Http::Respo
 void UserApiImpl::get_user_by_name(const std::string &username, Pistache::Http::ResponseWriter &response) {
     response.send(Pistache::Http::Code::Ok, "Do some magic\n");
 }
-void UserApiImpl::login_user(const Pistache::Optional<std::string> &username, const Pistache::Optional<std::string> &password, Pistache::Http::ResponseWriter &response) {
+void UserApiImpl::login_user(const std::optional<std::string> &username, const std::optional<std::string> &password, Pistache::Http::ResponseWriter &response) {
     response.send(Pistache::Http::Code::Ok, "Do some magic\n");
 }
 void UserApiImpl::logout_user(Pistache::Http::ResponseWriter &response) {

--- a/samples/server/petstore/cpp-pistache/impl/UserApiImpl.h
+++ b/samples/server/petstore/cpp-pistache/impl/UserApiImpl.h
@@ -24,10 +24,10 @@
 #include <pistache/http.h>
 #include <pistache/router.h>
 #include <memory>
+#include <optional>
 
 #include <UserApi.h>
 
-#include <pistache/optional.h>
 
 #include "User.h"
 #include <string>
@@ -48,7 +48,7 @@ public:
     void create_users_with_list_input(const std::vector<User> &body, Pistache::Http::ResponseWriter &response);
     void delete_user(const std::string &username, Pistache::Http::ResponseWriter &response);
     void get_user_by_name(const std::string &username, Pistache::Http::ResponseWriter &response);
-    void login_user(const Pistache::Optional<std::string> &username, const Pistache::Optional<std::string> &password, Pistache::Http::ResponseWriter &response);
+    void login_user(const std::optional<std::string> &username, const std::optional<std::string> &password, Pistache::Http::ResponseWriter &response);
     void logout_user(Pistache::Http::ResponseWriter &response);
     void update_user(const std::string &username, const User &body, Pistache::Http::ResponseWriter &response);
 

